File: app/src/main/java/app/revanced/integrations/settings/PlayerType.java
package app.revanced.integrations.settings;

public enum PlayerType {

    NONE,
    HIDDEN,
    WATCH_WHILE_MINIMIZED,
    WATCH_WHILE_MAXIMIZED,
    WATCH_WHILE_FULLSCREEN,
    WATCH_WHILE_SLIDING_MAXIMIZED_FULLSCREEN,
    WATCH_WHILE_SLIDING_MINIMIZED_MAXIMIZED,
    WATCH_WHILE_SLIDING_MINIMIZED_DISMISSED,
    WATCH_WHILE_SLIDING_FULLSCREEN_DISMISSED,
    INLINE_MINIMAL,
    VIRTUAL_REALITY_FULLSCREEN,
    WATCH_WHILE_PICTURE_IN_PICTURE;

    /* renamed from: a */
    public final boolean m33524a() {
        return !m33520e() && m33523b() && m33517h();
    }

    /* renamed from: b */
    public final boolean m33523b() {
        return this == WATCH_WHILE_FULLSCREEN || this == VIRTUAL_REALITY_FULLSCREEN || this == WATCH_WHILE_PICTURE_IN_PICTURE;
    }

    /* renamed from: c */
    public final boolean m33522c() {
        return this == NONE || m33521d();
    }

    /* renamed from: d */
    public final boolean m33521d() {
        return this == INLINE_MINIMAL;
    }

    /* renamed from: e */
    public final boolean m33520e() {
        return this == WATCH_WHILE_PICTURE_IN_PICTURE;
    }

    /* renamed from: f */
    public final boolean m33519f() {
        return (this == NONE || this == HIDDEN) ? false : true;
    }

    /* renamed from: g */
    public final boolean m33518g() {
        return this == VIRTUAL_REALITY_FULLSCREEN;
    }

    /* renamed from: h */
    public final boolean m33517h() {
        return this == WATCH_WHILE_MINIMIZED || this == WATCH_WHILE_MAXIMIZED || this == WATCH_WHILE_FULLSCREEN || this == WATCH_WHILE_SLIDING_MAXIMIZED_FULLSCREEN || this == WATCH_WHILE_SLIDING_MINIMIZED_MAXIMIZED || this == WATCH_WHILE_SLIDING_MINIMIZED_DISMISSED || this == WATCH_WHILE_SLIDING_FULLSCREEN_DISMISSED || this == WATCH_WHILE_PICTURE_IN_PICTURE;
    }

    /* renamed from: i */
    public final boolean m33516i() {
        return this == WATCH_WHILE_MAXIMIZED || this == WATCH_WHILE_FULLSCREEN;
    }

    /* renamed from: j */
    public final boolean m33515j() {
        return m33516i() || this == WATCH_WHILE_SLIDING_MAXIMIZED_FULLSCREEN;
    }

    /* renamed from: k */
    public final boolean m33514k() {
        return this == WATCH_WHILE_MINIMIZED || this == WATCH_WHILE_SLIDING_MINIMIZED_DISMISSED;
    }

    /* renamed from: l */
    public final boolean m33513l() {
        return m33514k() || m33512m();
    }

    /* renamed from: m */
    public final boolean m33512m() {
        return this == WATCH_WHILE_SLIDING_MINIMIZED_MAXIMIZED || this == WATCH_WHILE_SLIDING_MINIMIZED_DISMISSED || this == WATCH_WHILE_SLIDING_MAXIMIZED_FULLSCREEN || this == WATCH_WHILE_SLIDING_FULLSCREEN_DISMISSED;
    }

}

File: app/src/main/java/app/revanced/integrations/shared/PlayerType.kt
package app.revanced.integrations.shared

import app.revanced.integrations.utils.Event

/**
 * WatchWhile player type
 */
@Suppress("unused")
enum class PlayerType {
    NONE, // includes Shorts and Stories playback
    HIDDEN, // A Shorts or Stories, if a regular video is minimized and a Short/Story is then opened
    WATCH_WHILE_MINIMIZED,
    WATCH_WHILE_MAXIMIZED,
    WATCH_WHILE_FULLSCREEN,
    WATCH_WHILE_SLIDING_MAXIMIZED_FULLSCREEN,
    WATCH_WHILE_SLIDING_MINIMIZED_MAXIMIZED,
    WATCH_WHILE_SLIDING_MINIMIZED_DISMISSED, // When opening a short while a regular video is opened, the type can momentarily be this
    WATCH_WHILE_SLIDING_FULLSCREEN_DISMISSED,
    INLINE_MINIMAL, // home feed video playback
    VIRTUAL_REALITY_FULLSCREEN,
    WATCH_WHILE_PICTURE_IN_PICTURE;

    companion object {
        /**
         * safely parse from a string
         *
         * @param name the name to find
         * @return the enum constant, or null if not found
         */
        @JvmStatic
        fun safeParseFromString(name: String): PlayerType? {
            return values().firstOrNull { it.name == name }
        }

        /**
         * the current player type, as reported by [app.revanced.integrations.patches.PlayerTypeHookPatch.YouTubePlayerOverlaysLayout_updatePlayerTypeHookEX]
         */
        @JvmStatic
        var current
            get() = currentPlayerType
            set(value) {
                currentPlayerType = value
                onChange(currentPlayerType)
            }
        @Volatile // value is read/write from different threads
        private var currentPlayerType = NONE

        /**
         * player type change listener
         */
        @JvmStatic
        val onChange = Event<PlayerType>()
    }

    /**
     * Check if the current player type is [NONE] or [HIDDEN]
     *
     * Does not include the first second after a short is opened when a regular video was already on screen.
     * To include that situation instead use [isNoneHiddenOrDismissed].
     */
    fun isNoneOrHidden(): Boolean {
        return this == NONE || this == HIDDEN
    }

    /**
     * Check if the current player type is [NONE], [HIDDEN], or [WATCH_WHILE_SLIDING_MINIMIZED_DISMISSED]
     *
     * Useful to check if a short is playing.
     *
     * @return If nothing, a Short, a Story,
     *         _or a regular minimized video is sliding off screen to a dismissed or hidden state_
     */
    fun isNoneHiddenOrDismissed(): Boolean {
        return this == NONE || this == HIDDEN || this == WATCH_WHILE_SLIDING_MINIMIZED_DISMISSED
    }

}